drivers/focuser/pegasus_prodigyMF.cpp

```diff
@@ -0,0 +1,552 @@
+    }
+    else
+    {
+        deleteProperty(MaxSpeedNP.name);
+        deleteProperty(TemperatureNP.name);
+        deleteProperty(FirmwareVersionTP.name);
+    }
+
+    return true;
+}
+
+bool PegasusProdigyMF::Handshake()
+{
+    if (ack())
+    {
+        LOGF_INFO("%s is online. Getting focus parameters...", this->getDeviceName());
+        return true;
+    }
+
+    LOGF_INFO(
+        "Error retrieving data from %s, please ensure device is powered and the port is correct.", this->getDeviceName());
+    return false;
+}
+
+const char *PegasusProdigyMF::getDefaultName()
+{
+    return "Pegasus ProdigyMF";
+}
+
+bool PegasusProdigyMF::ack()
+{
+    int nbytes_written = 0, nbytes_read = 0, rc = -1;
+    char errstr[MAXRBUF];
+    char cmd[2] = {0};
+    char res[16] = {0};
+
+    cmd[0] = '#';
+    cmd[1] = 0xA;
+
+    LOGF_DEBUG("CMD <%#02X>", cmd[0]);
+
+    tcflush(PortFD, TCIOFLUSH);
+
+    if ((rc = tty_write(PortFD, cmd, 2, &nbytes_written)) != TTY_OK)
+    {
+        tty_error_msg(rc, errstr, MAXRBUF);
+        LOGF_ERROR("Ack error: %s.", errstr);
+        return false;
+    }
+
+    if ((rc = tty_read_section(PortFD, res, 0xA, DMFC_TIMEOUT, &nbytes_read)) != TTY_OK)
+    {
+        tty_error_msg(rc, errstr, MAXRBUF);
+        LOGF_ERROR("Ack error: %s.", errstr);
+        return false;
+    }
+
+    // Get rid of 0xA
+    res[nbytes_read - 1] = 0;
+
+
+    if( res[nbytes_read - 2] == '\r') res[nbytes_read - 2] = 0;
+
+    LOGF_DEBUG("RES <%s>", res);
+
+    tcflush(PortFD, TCIOFLUSH);
+
+    if(strstr(res, "OK_PRDG") != nullptr)
+        return true;
+
+    return false;
+}
+
+
+bool PegasusProdigyMF::SyncFocuser(uint32_t ticks)
+{
+    int nbytes_written = 0, rc = -1;
+    char cmd[16] = {0};
+
+    snprintf(cmd, 16, "W:%ud", ticks);
+    cmd[strlen(cmd)] = 0xA;
+
+    LOGF_DEBUG("CMD <%s>", cmd);
+
+    // Set Position
+    if ((rc = tty_write(PortFD, cmd, strlen(cmd), &nbytes_written)) != TTY_OK)
+    {
+        char errstr[MAXRBUF];
+        tty_error_msg(rc, errstr, MAXRBUF);
+        LOGF_ERROR("sync error: %s.", errstr);
+        return false;
+    }
+
+    this->ignoreResponse();
+
+    return true;
+}
+
+bool PegasusProdigyMF::move(uint32_t newPosition)
+{
+    int nbytes_written = 0, rc = -1;
+    char cmd[16] = {0};
+
+    snprintf(cmd, 16, "M:%ud", newPosition);
+    cmd[strlen(cmd)] = 0xA;
+
+    LOGF_DEBUG("CMD <%s>", cmd);
+
+    // Set Position
+    if ((rc = tty_write(PortFD, cmd, strlen(cmd), &nbytes_written)) != TTY_OK)
+    {
+        char errstr[MAXRBUF];
+        tty_error_msg(rc, errstr, MAXRBUF);
+        LOGF_ERROR("move error: %s.", errstr);
+        return false;
+    }
+
+    this->ignoreResponse();
+
+    return true;
+}
+
+bool PegasusProdigyMF::ISNewSwitch(const char *dev, const char *name, ISState *states, char *names[], int n)
+{
+    if (dev != nullptr && strcmp(dev, getDeviceName()) == 0)
+    {
+
+    }
+    return INDI::Focuser::ISNewSwitch(dev, name, states, names, n);
+}
```

Comment:
No need to override if not in use.

Reply:
This is the initial driver to help out customers use the driver to focus. The switch is required (to be implemented in next revision) as the device has a USB and Power Hub.